powerbi-visual-builder/src_visual/visual.ts

```diff
@@ -519,18 +614,47 @@ namespace powerbi.extensibility.visual {
                 this.selectionManager.clear();
               }
             });
-            if (this.enableTooltip && this.host.tooltipService.enabled()) {
-              const service = this.host.tooltipService;
 
+            if (this.enableDrillDown) {
+              const service = this.selectionManager;
+              this.chartContainer.addContextMenuListener(
+                // TODO change to point object
+                (table, rowIndices, options) => {
+                  const { clientX, clientY, event } = options;
+                  if ((service as any).showContextMenu) {
+                    const selection = rowIndex2selectionID.get(rowIndices[0]);
+                    (service as any).showContextMenu(selection, {
+                      x: clientX,
+                      y: clientY
+                    });
+                    event.preventDefault();
+                  }
+                }
+              );
+            }
+            const powerBITooltips = dataset.tables.find(
+              table => table.name === "powerBITooltips"

```

Comment:
It might be helpful to pull out the string literal `"powerBITooltips"` into its own module level const

Reply:
and replace all its uses in here.

Reply:
Updated